pkg/compactor/shard_aware_deduplicate_filter.go

```diff
@@ -227,14 +236,14 @@ func (b *blockWithSuccessors) addSuccessorIfPossible(other *blockWithSuccessors)
 	return true
 }
 
-// isFullyIncludedInSuccessors returns true if this block is *fully* included in successor blocks.
-// This is true under following conditions:
+// isFullyIncludedInSuccessors returns true if this block is *fully* included in its own successor blocks.
+// This is true under the following conditions:
 //
-// - if this block has non-empty shardID and has *any* successors, then it is fully included already.
+// - if this block has a non-empty shardID, and it has *any* successors, then the block is fully included in its successors.
 //
-// - if this block doesn't have shardID, and there is any successor without shardID, then it if fully included in that successor.
+// - if this block doesn't have a shardID, and it has a successor without a shardID, then the block is fully included in its successors.
 //
-// - if this block doesn't have shardID, but ALL successors do, and together they cover all shards, then this block is fully included in successors.
+// - if this block doesn't have shardID, but *all* its successors do, and together they cover all shards, then the block fully included in its successors.
```

Comment:
I personally think this reads the best:
```suggestion
// - if this block doesn't have shardID, but *all* its successors do, and together they cover all shards, then it is fully included in its successors.
```